### DISCUSSION TITLE
Display new words as the user is typing rather than the whole paragraph

### DISCUSSION THREAD

**Discussion by ghost** [monkeytypegame/monkeytype#2326] [Ideas] [answered]
I found it a little bit distracting to have the whole paragraph displayed, I feel like it would be better if a new word shows up every time a word is correctly typed. That doesn't mean that the initial state should be one word, It'd rather be something like five words  (because that's around how far we think when we're typing in general), so at any point in time, there is gonna be five words that are untyped, with the set of words that are typed correctly/highlighted.

**Comment by Miodec** [comment, marked as answer]
Try disabling `show all lines`

**Reply by ghost** [reply]
It is disabled by default. Also, I was referring to words not lines.

**Reply by Miodec** [reply]
Is showing 3 lines too much?

**Reply by ghost** [reply]
It depends what the goal is. If the site is made for competing then I guess it is completely fine to have that. But, I look at it from "making-progress" point of view. Realistically, I would be thinking and typing. My thoughts are usually five to ten words long. So if I'm getting like 5 words at a time, I can memorize them easily. Also, I think a slight improvement in that regard, I think when we're typing to pick up speed, I don't think we care that much about the content (especially if it is random words) so it would be a little better if the words that are typed correctly just go off the screen as the new ones append to the end of the line, you can even have that whole typing happening in one line, no matter how fast the typist is.

**Reply by Miodec** [reply]
Oh right, having everything in one line. Would require a large rewrite of the typing screen, plus i feel like having the words come to you (and keeping your vision focused on one point) doesn't accurately reflect real typing either.

**Reply by ghost** [reply]
I agree. But when you're typing something you know for sure, you don't move your eyes a lot either. It's not really about the eyes (at least imo). The distraction factor that I mention is just a mental thing and has to do with "being a bit overwhelmed", it feels like it's never ending, whereas if you have one line and a few words, it's like "I will get them done".
I hope I made myself clear. This might be a personal feeling idk, what do you think?

**Reply by Miodec** [reply]
Have you tried the `plus one` and `plus two` funboxes?

**Reply by ghost** [reply]
Oh wow, I wasn't aware of this feature. I have just tried `plus two` and it's just what the doctor ordered. Thanks for the consistent responses. Cheers!
